http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -266,33 +266,61 @@ class ClusterMonitorActor(val cluster: Cluster,
     logger.info(s"Cluster ${cluster.projectNameString} has been stopped.")
 
     for {
-      clusterStatus <- getDbClusterStatus
+      cluster <- getDbCluster
       // create or update instances in the DB
       _ <- persistInstances(instances)
       // this sets the cluster status to stopped and clears the cluster IP
       _ <- dbRef.inTransaction { _.clusterQuery.updateClusterStatus(cluster.id, ClusterStatus.Stopped) }
       // reset the time at which the kernel was last found to be busy
       _ <- dbRef.inTransaction {_.clusterQuery.clearKernelFoundBusyDate(cluster.id)}
       // Record metrics in NewRelic
-      _ <- recordMetrics(clusterStatus, ClusterStatus.Stopped).unsafeToFuture()
+      _ <- recordMetrics(cluster.status, ClusterStatus.Stopped).unsafeToFuture()
     } yield ShutdownActor(RemoveFromList(cluster))
   }
 
-  private def checkCluster: Future[ClusterMonitorMessage] = {
-    getDbClusterStatus.flatMap {
-      case status if status.isMonitored =>
-        checkClusterInGoogle(status)
-      case status =>
-        logger.info(s"Stopping monitoring of cluster ${cluster.projectNameString} in status ${status}")
-        Future.successful(ShutdownActor(RemoveFromList(cluster)))
+  private def createClusterInGoogle: Future[ClusterMonitorMessage] = {
+    logger.info(s"Attempting to create cluster ${cluster.projectNameString} in Google...")
+    val future = for {
+      clusterResult <- clusterHelper.createGoogleCluster(cluster).unsafeToFuture()
+      (cluster, initBucket, saKey) = clusterResult
+      _ <- dbRef.inTransaction { _.clusterQuery.updateAsyncClusterCreationFields(Option(GcsPath(initBucket, GcsObjectName(""))), saKey, cluster) }
+      _ = logger.info(s"Cluster ${cluster.projectNameString} was successfully created. Will monitor the creation process.")
+    } yield NotReadyCluster(cluster.status, Set.empty)
+
+    future.recover { case e =>
+      logger.error(s"Failed to create cluster ${cluster.projectNameString} in Google", e)
+      val errorMessage = e match {
+        case leoEx: LeoException =>
+          ErrorReport.loggableString(leoEx.toErrorReport)
+        case _ =>
+          s"Failed to create cluster ${cluster.projectNameString} due to ${e.toString}"
+      }
+      FailedCluster(ClusterErrorDetails(-1, Option(errorMessage)), Set.empty)
```

Comment:
any reason why `Option` is used in many places? Looks to me it can just be `Some` when the variable is never going to be `null`

Reply:
Habit - I agree to use `Some` when not dealing with potential nulls